### DISCUSSION TITLE
why don't we prefer the macOS alias ?

### DISCUSSION THREAD

**Discussion by jackywu** [ololx/quick-symlink#23] [Announcements]
this project is very good for me. thanks a lot.

I followed the installation instruction on main page and get failed on macOS sonama 14.5. do you know whether this app will work well in this OS version ?

just another question here, why don't we prefer the macOS aliases over symlink ? do you know anythings aliases can't do but symlink can do ? 

thanks.

**Comment by ololx** [comment]
Hello @jackywu,
Thank you for this issue. 

I'm very sorry that you couldn't install this application following the instructions. I probably need to update them soon. I'll try to help you and answer your questions.

#1 Question
Yes, QuickSymlink works on macOS Sonoma. I have version 14.2.1 installed. Unfortunately, I don't have enough details about your issue, so I can't give you an exact answer on what you need to do to install it in your case. But I tried uninstalling and reinstalling it on my system and prepared the following steps for you (which I'll later update on the main page):

1. Download the latest release version of the application from GitHub using this link: [https://github.com/ololx/quick-symlink/releases/download/v.0.10.5/quick-symlink.app.zip](https://github.com/ololx/quick-symlink/releases/download/v.0.10.5/quick-symlink.app.zip)
2. Locate the downloaded file (macOS will likely unzip it and place it in the Downloads folder). If the file is still in .zip format, unzip it.
3. Copy the file to your Applications directory (as you would with other mac applications).
4. Try to run it (double-click the quick-symlink.app file in the Applications directory or single-click it in the application launcher). macOS might display a warning message saying:
   ```
   "quick-symlink.app" can't be opened because Apple cannot check it for malicious software.
   This software needs to be updated.
   Contact the developer for more information.
   ```
<img width="266" alt="Screenshot 2024-07-04 at 8 46 46 AM" src="https://github.com/ololx/quick-symlink/assets/57698089/0a8b6b2a-5a51-4704-a01f-40bfda394395">

   To fix this, go to Settings -> Privacy & Security, scroll down to the Security section, and near the message
   ```
   "quick-symlink.app" was blocked from use because it is not from an identified developer.
   ```
   click the Open Anyway button.
   
<img width="711" alt="Screenshot 2024-07-04 at 8 58 31 AM" src="https://github.com/ololx/quick-symlink/assets/57698089/bbf5c786-a032-41f4-83c3-16c5a2d57158">
<img width="493" alt="Screenshot 2024-07-04 at 8 47 28 AM" src="https://github.com/ololx/quick-symlink/assets/57698089/2059642d-9243-4589-8e82-2c72efd7e326">
   
5. After this, try launching the application again.
6. Once you launch the application, it will open. You can then close it; at this point, all extensions should be installed, and you can use them through the context menu as shown in the examples. If the extensions don't appear in the context menu, you need to check Settings -> Extensions, find the Added extensions section, and select (check) quick-symlink.

<img width="480" alt="Screenshot 2024-07-04 at 8 49 54 AM" src="https://github.com/ololx/quick-symlink/assets/57698089/8461ce36-4b57-47d9-bd18-657d431c9c3b">
<img width="479" alt="Screenshot 2024-07-04 at 8 49 40 AM" src="https://github.com/ololx/quick-symlink/assets/57698089/539af67e-b977-4fee-a79a-3e76316c396e">

7. It is best to immediately restart Finder: Force Quit... -> Finder -> Relaunch.

<img width="310" alt="Screenshot 2024-07-04 at 9 04 17 AM" src="https://github.com/ololx/quick-symlink/assets/57698089/fc28af5b-6b9a-43b4-947e-fd740d8e3079">

#2 Question

Key differences:

1. **Portability**: Symbolic links are more portable. They work on most Unix-like operating systems, such as Linux, and can be used in scripts and other automated processes. Aliases are specific to macOS and may not work on other platforms.

2. **Functionality**: Symbolic links can point to files and directories on different file systems and even network locations. Aliases mainly work within the same file system and often depend on the specific location of files.

3. **File Type**:
   - An alias is a regular file with metadata.
   - A symbolic link is a special type of file that points to the path of another file or directory.

4. **Size**:
   - An alias is larger because it stores more information (metadata).
   - A symbolic link has a size equal to the length of the path it references.

5. **Display in Output**:
   - An alias appears as a regular file in a file list.
   - A symbolic link clearly shows what it points to using the "->" symbol.

6. **Behavior when the Target File/Directory is Moved or Deleted**:
   - A symbolic link becomes invalid if the target directory is moved or deleted.
   - An alias in macOS can attempt to locate the moved file/directory using a unique identifier, making it more resilient to changes in location.

**Reply by Jingyuan-Zheng** [reply]
It doesn't work on macOS 15 Sequoia. In Extensions, Quick Symlink can not be find after restarting or installing by terminal manually...

**Comment by jackywu** [comment]
@ololx thanks for your detailed explanation. cool.
